crates/re_viewport/src/space_view_entity_picker.rs

```diff
@@ -188,35 +188,25 @@ fn add_entities_line_ui(
         });
 
         ui.with_layout(egui::Layout::right_to_left(egui::Align::Center), |ui| {
-            // TODO(emilk): Show at most one button.
+            if entity_path_filter.contains_rule_for_exactly(entity_path) {
+                // Reset-button
+                // Shows when an entity is explicitly excluded or included
+                let response = ctx.re_ui.small_icon_button(ui, &re_ui::icons::RESET);
 
-            // Reset-button
-            {
-                let enabled = add_info.can_add_self_or_descendant.is_compatible()
-                    && entity_path_filter.contains_rule_for_exactly(entity_path);
-
-                ui.add_enabled_ui(enabled, |ui| {
-                    let response = ctx.re_ui.small_icon_button(ui, &re_ui::icons::RESET);
-
-                    if response.clicked() {
-                        space_view.remove_filter_rule_for(ctx, &entity_tree.path);
-                    }
-
-                    if enabled {
-                        if is_explicitly_excluded {
-                            response.on_hover_text("Stop excluding this EntityPath.");
-                        } else if is_explicitly_included {
-                            response.on_hover_text("Stop including this EntityPath.");
-                        }
-                    }
-                });
-            }
+                if response.clicked() {
+                    space_view.remove_filter_rule_for(ctx, &entity_tree.path);
+                }
 
-            // Remove-button
-            {
-                let enabled = is_included
-                    && add_info.can_add_self_or_descendant.is_compatible()
-                    && query_result.contains_any(&entity_tree.path);
+                if is_explicitly_excluded {
+                    response.on_hover_text("Stop excluding this EntityPath.");
+                } else if is_explicitly_included {
+                    response.on_hover_text("Stop including this EntityPath.");
+                }
+            } else if is_included {
+                // Remove-button
+                // Shows when an entity is already included (but not explicitly)
+                // Only enabled if the entity is compatible.
```

Comment:
When would this ever be disabled?

It is included, yet not compatible? That isn't really possible, unless its compatibility has changed since it was added?

Reply:
It shows up in the case that an incompatible entity is included via a recursive include from a parent. For consistency with the Add UI, I was disabling the button, because that hover is where we currently show the explanation as to **why** it can't be included... but I guess there's no reason not to. It's still valid to exclude an incompatible thing.